xonsh/color_tools.py

```diff
@@ -397,17 +397,19 @@ def color_dist(x, y):
     return math.sqrt((x[0]-y[0])**2 + (x[1]-y[1])**2 + (x[2]-y[2])**2)
 
 
-def find_closest_color(x, pallette):
-    return min(sorted(pallette.keys())[::-1],
-               key=lambda k: color_dist(x, pallette[k]))
+def find_closest_color(x, palette):
+    return min(sorted(palette.keys())[::-1],
+               key=lambda k: color_dist(x, palette[k]))
 
 
-def make_pallete(strings):
-    """Makes a color pallete from a colection of strings."""
-    pallette = {}
+def make_palette(strings):
+    """Makes a color palette from a collection of strings."""
+    palette = {}
     for s in strings:
         while '#' in s:
             _, t = s.split('#', 1)
             t, _, s = t.partition(' ')
-            pallette[t] = rgb_to_ints(t)
-    return pallette
+            palette[t] = rgb_to_ints(t)
+    return palette
+
+make_pallette = make_palette
```

Comment:
How does this look?

```python
import functools
import warnings


def deprecated(deprecated_in=None, removed_in=None):
    """Deprecate a function in a graceful manner.

    Updates the decorated function's docstring to mention the version
    that deprecation occurred in and the version it will be removed
    in if both of these values are passed.

    In all circumstances, call ``warnings.warn`` with details, while
    raising ``DeprecationWarning``.

    Usage:

        >>> @deprecated(deprecated_in='0.5.10', removed_in='0.6.0')
        ... def my_function(thing1, thing2):
        ...     print(thing1, thing2)
        >>> my_function('hello', 'world')
        hello world
        >>> ('my_function has been deprecated in version 0.5.10 and will be '
        ...  'removed in version 0.6.0' == my_function.__doc__)
        True
    """
    message_suffix = _message_suffix(deprecated_in, removed_in)
    if not message_suffix:
        message_suffix = ''

    def decorated(func):
        warning_message = '{} has been deprecated'.format(func.__name__)
        warning_message += message_suffix

        @functools.wraps(func)
        def wrapped(*args, **kwargs):
            func(*args, **kwargs)
            warnings.warn(warning_message, DeprecationWarning)

        wrapped.__doc__ = (
            wrapped.__doc__ + '\n\n' + warning_message
            if wrapped.__doc__ else warning_message)

        return wrapped
    return decorated


def _message_suffix(deprecated_in, removed_in):
    if deprecated_in and removed_in:
        message_suffix = (
            ' in version {} and will be removed in version {}'.format(
                deprecated_in, removed_in))
    elif deprecated_in and not removed_in:
        message_suffix = (
            ' in version {}'.format(deprecated_in))
    elif not deprecated_in and removed_in:
        message_suffix = (
            ' and will be removed in version {}'.format(removed_in))
    else:
        message_suffix = None

    return message_suffix
```